sdk/eventhub/Azure.Messaging.EventHubs.Processor/src/Storage/BlobsCheckpointStore.cs

```diff
@@ -235,6 +291,11 @@ await foreach (BlobItem blob in ContainerClient.GetBlobsAsync(traits: BlobTraits
                 }
 
                 return checkpoints;
+            };
+
+            try
+            {
+                return ApplyRetryPolicy(listCheckpointsAsync, cancellationToken);
```

Comment:
Should we be retrying?   Doesn't the blob client handle this on our behalf?

Reply:
Just answered the same question in the comment above.